torch/_inductor/codegen/cpp.py

```diff
@@ -429,43 +496,25 @@ def merge_outer_fusion_kernels(
             kernel.loop_nest for kernel in cpp_kernel_proxy_list
         ]
         kernel_group = cpp_kernel_proxy_list[0].kernel_group
+        outer_loop_fused_kernel = OuterLoopFusedKernel(kernel_group)
+        outer_loop_fused_kernel.inner = [
+            proxy.loop_nest for proxy in cpp_kernel_proxy_list
+        ]
+        outer_loop_fused_kernel.fusion_depth = self.outer_loop_fusion_depth
 
-        def _merge_outer_fusion_loop_levels(
-            loop_level_nested_list: List[List["LoopLevel"]],
-            outer_loop_fusion_depth,
-        ):
-            assert outer_loop_fusion_depth >= 1
-            # Assume no main/tail loop split at any outer loop fusion depth
-            assert all(
-                len(loop_level_list) == 1 for loop_level_list in loop_level_nested_list
-            )
-            if (outer_loop_fusion_depth := outer_loop_fusion_depth - 1) >= 1:
-                # Further merge the next loop level
-                next_loop_level_nested_list = [
-                    loop_level_list[0].inner
-                    for loop_level_list in loop_level_nested_list
-                ]
-                _merge_outer_fusion_loop_levels(
-                    next_loop_level_nested_list,
-                    outer_loop_fusion_depth,
-                )
-            else:
-                outer_loop_fused_kernel = OuterLoopFusedKernel(kernel_group)
-                loop_level_of_first_kernel = loop_level_nested_list[0][0]
-                for kernel_idx in range(len(loop_level_nested_list)):
-                    outer_loop_fused_kernel.inner.append(
-                        deepcopy(loop_level_nested_list[kernel_idx][0]),
-                    )
-                loop_level_of_first_kernel.inner = []
-                loop_level_of_first_kernel.kernel = outer_loop_fused_kernel
-
-        # Merge the List[LoopNestWithSplit] from cpp_kernel_proxy_list
-        # into cpp_kernel_proxy_list[0].loop_nest
-        _merge_outer_fusion_loop_levels(
-            [_loop_nest.root for _loop_nest in loop_nest_list],  # type: ignore[misc]
-            self.outer_loop_fusion_depth,
-        )
-        return cpp_kernel_proxy_list[0]
+        def find_deepest_proxy(proxy_list):
+            deepest_proxy = None
+            depth = -1
+            for proxy in proxy_list:
+                if len(proxy.loop_nest.loops) > depth:
+                    depth = len(proxy.loop_nest.loops)
+                    deepest_proxy = proxy
+            assert deepest_proxy is not None
+            return deepest_proxy
+
+        deepest_proxy = find_deepest_proxy(cpp_kernel_proxy_list)
```

Comment:
Why we have to find the deepest kernel proxy?

Reply:
We do not need it now since we will let the `LoopNest` in `OuterFusedKernel` start from depth 0 and we do not need fusion `depth`.
Previously we will loss the LoopLevel to gen if we do not choose the deepest kernel proxy here.